xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/editBus.js

```diff
@@ -0,0 +1,232 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+
+/**
+ * Vue event bus for the edit action states.
+ * Sent events:
+ * - start-editing-entry({entryId, propertyId}): when an entry passes in edit mode
+ * - cancel-editing-entry({entryId, propertyId}): when an entry goes out of edit mode without saving its edits
+ * - save-editing-entry({entryId, propertyId, content}): when an entry goes out of edit model and wants to save its
+ *   edits. The content is an arbitrary value specific to the property's displayer type.
+ */
+define('edit-bus', ['vue'], (Vue) => {
+
+  var _logic = undefined;
+  var _editBus = new Vue();
+  var _editBusService = undefined;
+
+  /**
+   * Utility method to aggregate the provided content into a vals object.
+   * For each key of the content object, a corresponding key is created in vals. If the key is found only once, the
+   * corresponding value is assigned in vals. If the key is found more than once, an array of the found values is
+   * set for the corresponding key in vals.
+   * @param content the content to add to vals
+   * @param vals the object in which to aggregate the provided content.
+   */
+  function aggregate(content, vals) {
+    for (const key in content) {
+      const value = content[key];
```

Comment:
I don't get it. If you do ``content[key]`` then the ``key`` must be **unique** in ``content``, so the only way to have a "duplicated" key (but it's not actually duplicated) is if the key already exists in ``vals`` before this function is called, but it seems you're calling with an empty object below so you never have "duplicated" keys.

Reply:
You are right, this is not useful anymore. It used to be needed where I was updating a full line at first, but now that we update a single cell at a time it is not. I'll remove it.